### PR TITLE
Update ORT Performance threading docs

### PR DESCRIPTION
**Pull request by ivberg**
### Description
Update ORT Performance threading docs. Clearer defaults, clearer explanation of INTRA thread behavior. Document spinning. Provide links to APIs

### Motivation and Context
After discussion with ORT Performance team and in-depth testing of these settings. It was determined that docs were not clear, misleading at times, or needed missing info.

### CODE REVIEW

**Comment by ivberg** [comment]
For the checklinks failure that is not related to this PR. Also the URL seems to be valid that is claims to have hit a 404 - https://github.com/microsoft/onnx-converters-private/issues/new/choose

File: docs/performance/tune-performance/threading.md [hunk 1]
```diff
@@ -14,75 +14,105 @@ nav_order: 3
 {:toc}
 
 
-For the default CPU execution provider, you can use the following knobs in the Python API to control the thread number:
+For the default CPU execution provider, setting defaults are provided to get fast inference performance. You can customize the performance using the following knobs in the API to control the thread count and other settings:
```
Comment (pranavsharma) [inline docs/performance/tune-performance/threading.md:17]:
```suggestion
For the default CPU execution provider, defaults settings are provided to get fast inferencing performance. You can customize the performance using the following knobs in the API to control the thread count and other settings:
```

File: docs/performance/tune-performance/threading.md [hunk 2]
```diff
@@ -14,75 +14,105 @@ nav_order: 3
...
+    * E.g. 6-core machine (with 12 HT logical processors) = 6 total INTRA threads
 
-  * `sess_options.intra_op_num_threads = 2` controls the number of threads to use to run the model.
 
 * Sequential vs Parallel Execution
 
-  * `sess_options.execution_mode = rt.ExecutionMode.ORT_SEQUENTIAL` controls whether the operators in the graph run sequentially or in parallel. Usually when a model has many branches, setting this option to `ORT_PARALLEL` will provide better performance.
+  * Controls whether _multiple_ operators in the graph (_across_ nodes) run sequentially or in parallel. 
+  * Default: `sess_options.execution_mode = rt.ExecutionMode.ORT_SEQUENTIAL`
+  * Usually when a model has many branches, setting this option to `ORT_PARALLEL` will provide better performance. This could also hurt performance on some models without many branches.
```
Comment (pranavsharma) [inline docs/performance/tune-performance/threading.md:45]: ORT_PARALLEL is not supported anymore. You can safely omit this part and all references to INTER op threadpool.

File: docs/performance/tune-performance/threading.md [hunk 3]
```diff
@@ -14,75 +14,105 @@ nav_order: 3
...
 ```python
 sess_opt = SessionOptions()
 sess_opt.intra_op_num_threads = 3
 sess = ort.InferenceSession('model.onnx', sess_opt)
 ```
 
-With above configuration, two threads will be created in the pool, so along with the main calling thread, there will be three threads in total to participate in intra-op computation.
-By default, each session will create one thread per phyical core (except the 1st core) and attach the thread to that core.
-However, if customer explicitly set the number of threads like showcased above, there will be no affinity set to any of the created thread.
+With the above configuration of 3 total threads, two extra threads will be created in the addtional INTRA pool, so along with the main calling thread, there will be three threads in total to participate in intra-op computation. However, if customer explicitly set the number of threads like showcased above, there will be no affinity set to any of the created thread.
```
Comment (pranavsharma) [inline docs/performance/tune-performance/threading.md:76]:
```suggestion
With the above configuration of 3 total threads, two extra threads will be created in the INTRA pool, so along with the main calling thread, there will be three threads in total to participate in intra-op computation. However, if customer explicitly set the number of threads like showcased above, there will be no affinity set to any of the created thread.
```

**Review by pranavsharma — COMMENTED** [review]
Thanks for taking care of this.

**Review by ivberg — COMMENTED** [review]
Small fixes for PR review 1

**Comment by ivberg** [comment]
Preview of new doc - https://github.com/ivberg/onnxruntime/blob/ivberg-threading-perf-doc/docs/performance/tune-performance/threading.md

**Review by pranavsharma — APPROVED** [review]

File: docs/performance/tune-performance/threading.md (hunk 2 above)
Comment (ivberg) [reply docs/performance/tune-performance/threading.md:45]: Also I would prefer ORT team officially deprecates INTER in another PR. I think the scope of this PR is to make clearer the existing supported settings

**Comment by ivberg** [comment]
> For the checklinks failure that is not related to this PR. Also the URL seems to be valid that is claims to have hit a 404 - https://github.com/microsoft/onnx-converters-private/issues/new/choose

@pranavsharma How should we handle this? Can you force merge the PR OR should we move this OR do you have a better working URL?

**Comment by ivberg** [comment]
> > For the checklinks failure that is not related to this PR. Also the URL seems to be valid that is claims to have hit a 404 - https://github.com/microsoft/onnx-converters-private/issues/new/choose
> 
> @pranavsharma How should we handle this? Can you force merge the PR OR should we move this OR do you have a better working URL?

Gentle ping @pranavsharma

> > For the checklinks failure that is not related to this PR. Also the URL seems to be valid that is claims to have hit a 404 - https://github.com/microsoft/onnx-converters-private/issues/new/choose
> 
> @pranavsharma How should we handle this? Can you force merge the PR OR should we move this OR do you have a better working URL?

Gentle ping @pranavsharma

**Comment by hariharans29** [comment]
> > > For the checklinks failure that is not related to this PR. Also the URL seems to be valid that is claims to have hit a 404 - https://github.com/microsoft/onnx-converters-private/issues/new/choose
> > 
> > 
> > @pranavsharma How should we handle this? Can you force merge the PR OR should we move this OR do you have a better working URL?
> 
> Gentle ping @pranavsharma
> 
> > > For the checklinks failure that is not related to this PR. Also the URL seems to be valid that is claims to have hit a 404 - https://github.com/microsoft/onnx-converters-private/issues/new/choose
> > 
> > 
> > @pranavsharma How should we handle this? Can you force merge the PR OR should we move this OR do you have a better working URL?
> 
> Gentle ping @pranavsharma

Hi @natke - Can this PR be merged ? Also, I don't see a 404 for this URL (https://github.com/microsoft/onnx-converters-private/issues/new/choose) anymore, I guess that link is up and running now ?

**Review by natke — APPROVED** [review]

**Comment by natke** [comment]
Thank you @ivberg!

**Pull request merged**